pkg/config/config.go

```diff
@@ -55,19 +56,26 @@ func buildMainConfig(config *viper.Viper, ddconfigPath, ddconfdPath string) erro
 	config.SetDefault("skip_ssl_validation", false)
 	config.SetDefault("run_path", "/opt/datadog-agent/run")
 
+	hostname, err := os.Hostname()
 	if isAgent5 {
 		// for agent5, we don't want people to have to set log_enabled in the config
 		config.SetDefault("log_enabled", true)
 	} else {
 		config.SetDefault("log_enabled", false)
+		// agent6 has a util.GetHostname() that agent5 does not have
+		hostname, err = util.GetHostname()
```

Comment:
Very good point for thinking about doing this for Agent6 only.
I think we don't need to do that, as the agent6's logic falls back on using `Hostname`, and before applies some logic that makes sense even for agent5, so I extracted from it. But it's very important to keep that in mind, as you did!